packages/wonder-blocks-tooltip/components/tooltip-anchor.js

```diff
@@ -0,0 +1,168 @@
+
+    // Callback to be invoked when the anchored content is mounted.
+    // This provides a reference to the anchored content, which can then be
+    // used for calculating tooltip bubble positioning.
+    anchorRef: (?Element) => mixed,
+
+    // When true, the element wrapped by the anchor will be given tabindex=0
+    // to make it keyboard focusable. This value defaults to true. One might set
+    // this to false in circumstances where the wrapped component already can
+    // receive focus or contains an element that can.
+    // Use good judgement when overriding this value, the tooltip content should
+    // be accessible via keyboard in all circumstances where the tooltip would
+    // appear using the mouse, so very those use-cases.
+    forceAnchorFocusivity?: boolean,
+|};
+
+type State = {|
+    active: boolean,
+|};
+
+export default class TooltipAnchor extends React.Component<Props, State> {
+    _weSetFocusivity: ?boolean;
+    _anchorNode: ?Element;
+    _focused: boolean;
+    _hovered: boolean;
+
+    static defaultProps = {
+        forceAnchorFocusivity: true,
+    };
+
+    state = {
+        active: false,
+    };
+
+    componentDidMount() {
+        const anchorNode = ReactDOM.findDOMNode(this);
+
+        // This should never happen, but we have this check here to make flow
+        // happy and ensure that if this does happen, we'll know about it.
+        if (anchorNode instanceof Text) {
+            throw new Error(
+                "TooltipAnchor must be applied to an Element. Text content is not supported.",
+            );
+        }
+
+        this._anchorNode = anchorNode;
+        this._updateFocusivity();
+        if (anchorNode) {
+            // TODO(somewhatabstract): Detect ESC key to deactivate
+            // TODO(somewhatabstract): Work out how to allow pointer to go over
+            // the tooltip content to keep it active.
+
+            anchorNode.addEventListener("focusin", () => this._handleFocusIn());
+            anchorNode.addEventListener("focusout", () =>
+                this._handleFocusOut(),
+            );
+            anchorNode.addEventListener("mouseenter", () =>
+                this._handleMouseEnter(),
+            );
+            anchorNode.addEventListener("mouseleave", () =>
+                this._handleMouseLeave(),
+            );
+
+            this.props.anchorRef(this._anchorNode);
+        }
+    }
+
+    componentDidUpdate(prevProps: Props) {
+        if (
+            prevProps.forceAnchorFocusivity !==
+                this.props.forceAnchorFocusivity ||
+            prevProps.children !== this.props.children
+        ) {
+            this._updateFocusivity();
+        }
+    }
+
+    componentWillUnmount() {
+        const anchorNode = this._anchorNode;
+        if (anchorNode) {
+            anchorNode.removeEventListener("focusin", () =>
+                this._handleFocusIn(),
+            );
```

Comment:
Does this actually work?  I thought you had to pass in the exact same function when removing event listeners.

Reply:
I'll double check.